src/transformers/trainer_seq2seq.py

```diff
@@ -50,7 +55,7 @@ def __init__(
         model: Union["PreTrainedModel", nn.Module] = None,
         args: "TrainingArguments" = None,
         data_collator: Optional["DataCollator"] = None,
-        train_dataset: Optional[Dataset] = None,
+        train_dataset: Optional[Union[Dataset, IterableDataset, "datasets.Dataset"]] = None,
```

Comment:
As it's only here when type checking, please put it in between quotes so that it's known at runtime:

```suggestion
        train_dataset: Optional[Union[Dataset, "IterableDataset", "datasets.Dataset"]] = None,
```